ocf_data_sampler/torch_datasets/site.py

```diff
@@ -0,0 +1,271 @@
+            id=site_id,
+            x=site.longitude.values,
+            y=site.latitude.values,
+            coordinate_system="lon_lat"
+        )
+        locations.append(location)
+
+    return locations
+
+
+class SitesDataset(Dataset):
+    def __init__(
+            self,
+            config_filename: str,
+            start_time: str | None = None,
+            end_time: str | None = None,
+            gsp_ids: list[int] | None = None,
+    ):
+        """A torch Dataset for creating PVNet UK GSP samples
+
+        Args:
+            config_filename: Path to the configuration file
+            start_time: Limit the init-times to be after this
+            end_time: Limit the init-times to be before this
+            gsp_ids: List of GSP IDs to create samples for. Defaults to all
+        """
+
+        config = load_yaml_configuration(config_filename)
+
+        datasets_dict = get_dataset_dict(config)
+
+        # get all locations
+        self.locations = get_locations(datasets_dict['site'])
+
+        # Get t0 times where all input data is available
+        valid_t0_and_site_ids = find_valid_t0_and_site_ids(datasets_dict, config)
+
+        # Filter t0 times to given range
+
+        # Assign coords and indices to self
+        self.valid_t0_and_site_ids = valid_t0_and_site_ids
+
+        # Assign config and input data to self
+        self.datasets_dict = datasets_dict
+        self.config = config
+
+    def __len__(self):
+        return len(self.valid_t0_and_site_ids)
+
+    def _get_sample(self, t0: pd.Timestamp, location: Location) -> dict:
+        """Generate the PVNet sample for given coordinates
+
+        Args:
+            t0: init-time for sample
+            location: location for sample
+        """
+        sample_dict = slice_datasets_by_space(self.datasets_dict, location, self.config)
+        sample_dict = slice_datasets_by_time(sample_dict, t0, self.config)
+        sample_dict = compute(sample_dict)
+
+        sample = process_and_combine_datasets(sample_dict, self.config, t0, location, sun_position_key='site')
+
+        return sample
+
+    def get_location_from_site_id(self, site_id):
+        """Get location from system id"""
+
+        locations = [loc for loc in self.locations if loc.id == site_id]
+        if len(locations) == 0:
+            raise ValueError(f"Location not found for site_id {site_id}")
+
+        if len(locations) > 1:
+            logging.warning(f"Multiple locations found for site_id {site_id}, but will take the first")
+
+        return locations[0]
+
+    def __getitem__(self, idx):
+
+        # Get the coordinates of the sample
+        # TOD change to system ids
+        t0, site_id = self.valid_t0_and_site_ids.iloc[idx]
+
+        # get location from site id
+        location = self.get_location_from_site_id(site_id)
+
+        # Generate the sample
+        return self._get_sample(t0, location)
+
+    def get_sample(self, t0: pd.Timestamp, location: Location) -> dict:
```

Comment:
Should this be site_id instead of location here as an input? Otherwise I think this is just a duplicated function?

Reply:
I think this is a duplicate but it was inherited from uk regional, I think it also has two of the same

Reply:
I *think* only `_get_sample` is ever used but I migh be wrong. The origina also does two asserts which I think should go into `__getitem__` instead

Reply:
So in the uk_regional Dataset `get_sample` is a utility function for someone to check samples easily/test things are working locally and uses `gsp_id` as a parameter rather than a `location` (which is in `_get_sample` ) since that is what someone would easily be able to know/input and then it maps the id to a location in the function and calls `_get_sample` with that location. I believe the intention is the same here so `site_id` could be passed as an input parameter instead of `location` in `get_sample` for the same use case, would also need to add the site_id to location lookup

Reply:
Thanks, ive updated it to use site_id, good catch